tests/gradients/test_gradient_transform.py

```diff
@@ -289,6 +329,16 @@ def circuit(x, shots):
         ):
             qml.gradients.param_shift(circuit)(0.2, shots=100)
 
+    def test_trainable_args_detection_default(self):
+        """Test the fallback for the `_jacobian_trainable_args` method
+        of the gradient transform (not normally reached).

```

Comment:
How come the `else` branch was picked to be tested?

Reply:
See my [reply to your second comment](https://github.com/PennyLaneAI/pennylane/pull/1989#discussion_r766067962).